### PR TITLE
Added more Junit tests; Increase code quality 

### PR DESCRIPTION
**Pull request by franceneee**

### CODE REVIEW

**Review by Lusi711 — APPROVED** [review]

File: src/main/java/zeronote/notebooks/Notebook.java [hunk 1]
```diff
@@ -80,13 +80,14 @@ public void addSection(Section s) {
      */
     public int findSection(String searchKey) {
         int index = 0;
+        int notFound = -1;
```
Comment (NeilBaner) [inline src/main/java/zeronote/notebooks/Notebook.java]: Not sure this is necessary, but if at all, it should be a final, class-level field right?

File: src/main/java/zeronote/userinterface/InputParser.java [hunk 2]
```diff
@@ -43,17 +43,15 @@
...
-     *
      * @throws TaskTitleException               when the user's input does not include a task title.
      * @throws IncorrectDeadlineFormatException when the user's input does not include the DEADLINE_DELIMITER.
      */
     public String parseTaskTitle(String input) throws TaskTitleException, IncorrectDeadlineFormatException {
         if (input.startsWith(AddCommandTimetableMode.TASK_DELIMITER) && input.contains(
                 AddCommandTimetableMode.DEADLINE_DELIMITER)) {
             String taskTitle = input.substring(AddCommandTimetableMode.TASK_DELIMITER.length());
-            int indexPos = taskTitle.indexOf("/by");
+            int indexPos = taskTitle.indexOf(AddCommandTimetableMode.DEADLINE_DELIMITER);
```
Comment (NeilBaner) [inline src/main/java/zeronote/userinterface/InputParser.java:54]: nice

File: src/main/java/zeronote/notebooks/Notebook.java (hunk 1 above)
Comment (franceneee) [reply src/main/java/zeronote/notebooks/Notebook.java]: I just thought that returning -1 just like that would be an example of a magic literal xD that's why haha. so you mean I should initialise it right at the top, as a static final variable?

**Comment by NeilBaner** [comment]
I think that would be better, yes.

**Review by NeilBaner — APPROVED** [review]
Perfect, leggo.

**Pull request merged**